lib/puppetlabs_spec_helper/tasks/fixtures.rb

```diff
@@ -249,77 +272,46 @@ def current_thread_count(items)
 
   # returns the max_thread_count
   # because we may want to limit ssh or https connections
+  # @return [Integer] - current thread count
   def max_thread_limit
-    unless @max_thread_limit
-      # the default thread count is 10 but can be
-      # raised by using environment variable MAX_FIXTURE_THREAD_COUNT
-      @max_thread_limit = if ENV['MAX_FIXTURE_THREAD_COUNT'].to_i > 0
-                            ENV['MAX_FIXTURE_THREAD_COUNT'].to_i
-                          else
-                            10 # the default
-                          end
-    end
-    @max_thread_limit
+    # the default thread count is 10 but can be
+    # raised by using environment variable MAX_FIXTURE_THREAD_COUNT
+    @max_thread_limit ||= ENV['MAX_FIXTURE_THREAD_COUNT'].nil? ? 10 : ENV['MAX_FIXTURE_THREAD_COUNT'].to_i
```

Comment:
This reduces readability. I would suggest implementing the revised code in the more readable format used previously.